### PR TITLE
Cuevana new source added

### LINKED ISSUE #2024: Cuevana needs a domain change
### Source information

Cuevana Español 13.18

### Source language

Spanish

### Source new URL

https://wv1.cuevana3.eu

### Other details

Cuevana on the site above shows movies in good quality while the one currently is even lucky to show movies in 720, and the movies shown are in cam....can you use the link above? 

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated all installed extensions.
- [X] I have opened WebView and checked that the source URL is not updated yet.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by imper1aldev**
Checklist:
Closes #2024

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/es/cuevana/src/eu/kanade/tachiyomi/animeextension/es/cuevana/CuevanaEu.kt [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+            val preferredIdx = videoSorted.indexOfFirst { x -> x.quality == userPreferredQuality }
+            if (preferredIdx != -1) {
+                videoSorted.drop(preferredIdx + 1)
+                videoSorted[0] = videoSorted[preferredIdx]
+            }
+            videoSorted.toList()
+        } catch (e: Exception) {
+            this
+        }
+    }
```
Comment (bapeey) [inline src/es/cuevana/src/eu/kanade/tachiyomi/animeextension/es/cuevana/CuevanaEu.kt:253]: This doesn't work as expected, see: https://pl.kotl.in/scJEzSCVX

Comment (imper1aldev) [reply src/es/cuevana/src/eu/kanade/tachiyomi/animeextension/es/cuevana/CuevanaEu.kt]: I changed the current method to the method that the other extensions use to keep the same functionality.

**Comment by jmir1** [comment]
Can you explain what the difference is between cuevanach and cuevanaeu? Other than that they have different URLs.

**Comment by imper1aldev** [comment]
> Can you explain what the difference is between cuevanach and cuevanaeu? Other than that they have different URLs.

- CuevanaCh does not share the same html structure as CuevanaEu, so the same class cannot be used to extract the information from both sources.

- CuevanaCh is made in such a way that using Jsoup you can extract everything that is needed from the DOM, in the case of CuevanaEu, the information that is required is in JSON format in a "script" tag.

With these two points mentioned above, it can be known that the methodology of each method required in the extension would not be identical, so I made two different classes to separate the different codes and called them in the Factory.

**Comment by jmir1** [comment]
Yeah okay that makes sense but why do the sites have the same name? Do they have the exact same content or does one site have different movies/shows or maybe different video hosters/links?

**Comment by imper1aldev** [comment]
> Yeah okay that makes sense but why do the sites have the same name? Do they have the exact same content or does one site have different movies/shows or maybe different video hosters/links?

What you mention is what is happening, Cuevana has many copy sites, but since they do not belong to the same owner, each copy is programmed to look alike, but they are not the same in structure, only in style, in addition, each site has different content and different video servers.

**Comment by jmir1** [comment]
Okay thanks for explaining

**Review by jmir1 — APPROVED** [review]

**Pull request merged**